fix: update-open-sauced-goals-cache bug

TAKANOME-DEV opened the pull request:
<!--
  For Work In Progress Pull Requests, please use the Draft PR feature,
  see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.
  
  For a timely review/response, please avoid force-pushing additional
  commits if your PR already received reviews or comments.
  
  Before submitting a Pull Request, please ensure you've done the following:
  - 📖 Read the Open Sauced Contributing Guide: https://github.com/open-sauced/open-sauced/blob/HEAD/CONTRIBUTING.md#create-a-pull-request.
  - 📖 Read the Open Sauced Code of Conduct: https://github.com/open-sauced/open-sauced/blob/HEAD/CODE_OF_CONDUCT.md.
  - 👷‍♀️ Create small PRs. In most cases, this will be possible.
  - ✅ Provide tests for your changes.
  - 📝 Use descriptive commit messages.
  - 📗 Update any related documentation and include any relevant screenshots.
-->

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

<!-- Please do not leave this blank -->

This PR fixes the bug in `update-open-sauced-goals-cache.js`. If `open-sauced-goals` repo (issue name) do not conform to repo full name conventions, the goal caching action will fail.

## Related Tickets & Documents
<!-- 
Please use this format link issue numbers: Fixes #123
https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword 
-->
Fixes #38

## Mobile & Desktop Screenshots/Recordings

<!-- Visual changes require screenshots -->
This is my goals repo. As you can see, since the last 3 days the action keeps failing due to renovatebot `Dependency Dashboard` issue. But now everything looks good.

![goals_caching](https://user-images.githubusercontent.com/79809121/171294012-64b07c24-6172-40eb-a807-d6547d4fc739.png)


## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?



## [optional] What gif best describes this PR or how it makes you feel?



<!-- note: PRs with deletes sections will be marked invalid -->

bdougie commented:
Thanks for the PR @TAKANOME-DEV. Can you share what you changed in the description. At first glance it looks like an early return but it's all hidden amongst formatting changes.

TAKANOME-DEV commented:
This is what I added:
```diff
async function getRepoGoals(issues){
  return Promise.all(
    issues.map(async issue => {
      // all goal issues follow the "owner/repo" format
      let [owner, name] = issue.title.split("/");
+    if (!owner || !name) return;
    }
  )
}
```

mtfoley commented:
I think the early return will break production, I'd suggest filtering prior to the Promise.all.

@TAKANOME-DEV the issue you're seeing here is probably the result of the early return statement:

image from the bug you filed this morning:
![screenshot of console error](https://user-images.githubusercontent.com/79809121/171380903-e4752efd-6615-4402-88fb-b46f8cc70263.png)

affected app code: https://github.com/open-sauced/open-sauced/blob/4a26b94eaa1a1094ee0dd0d7ca76ead50da2acc4/src/components/RepositoryGoals.jsx#L52-L57

null where an object should be: https://github.com/TAKANOME-DEV/open-sauced-goals/blob/main/data.json#L8

TAKANOME-DEV commented:
Thanks for feedbacks I will work on it ASAP 👍

TAKANOME-DEV commented:
I made these changes:
```diff
async function getRepoGoals(issues){
  return Promise.all(
    issues.map(async issue => {
      // all goal issues follow the "owner/repo" format
      let [owner, name] = issue.title.split("/");
-    if (!owner || !name) return;
    }
  )
}

try {
  stagedIssues = await octokit.rest.issues.listForRepo({
    owner: login,
    repo: "open-sauced-goals",
  });
  console.log("stagedIssues", stagedIssues);
-   repoIssues = await octokit.paginate(stagedIssues);
+  const issues = await octokit.paginate(stagedIssues);
+  // filter issues that don't match "owner/name" format
+  repoIssues = issues.filter((repoIssue) => !repoIssue.title.match(/\s/) && !repoIssue.title.match(/^[^/]*$/));
} catch (err) {
  console.log(err);
}
```

TAKANOME-DEV commented:
Sorry about these commits I had some conflicts to resolve 😓

0-vortex commented:
> Sorry about these commits I had some conflicts to resolve 😓

No worries, we have squash pull requests for exactly that matter! 🍕
